app/src/main/java/com/lightricks/feedexercise/ui/feed/FeedViewModel.kt

```diff
@@ -21,12 +29,22 @@ open class FeedViewModel : ViewModel() {
 
     init {
         refresh()
+        feedItems = feedRepository.getFeedItems()
+        isEmpty.addSource(feedRepository.getFeedItems()){items -> isEmpty.postValue(items.isEmpty())}
     }
 
     fun refresh() {
-        //todo: fix the implementation
-        isLoading.value = false
-        isEmpty.value = true
+        isLoading.value = true
+        feedRepository.refresh()
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe({ isLoading.postValue(false) },{ throwable-> handleNetworkError(throwable)})
+    }
+
+    private fun handleNetworkError(error: Throwable?) {
+        isLoading.postValue(false)
+        networkErrorEvent.postValue(Event<String>(error?.message.toString()))
```

Comment:
Just make sure that there is a message even when the error is `null`. For example:
`error?.message.toString() ?: "error"`

Reply:
Try without toString:
`networkErrorEvent.value = Event(error?.message ?: "error")`